packages/wonder-blocks-core/components/clickable-behavior.js

```diff
@@ -269,13 +298,71 @@ export default class ClickableBehavior extends React.Component<
         this.dragging = false;
     }
 
+    async runCallbackAndMaybeNavigate(e: SyntheticEvent<>) {
+        const {
+            onClick = undefined,
+            beforeNav = undefined,
+            safeWithNav = undefined,
+            skipClientNav,
+            history,
+        } = this.props;
+        let shouldNavigate = true;
+
+        if (onClick) {
+            onClick(e);
+        }
+
+        try {
+            if (beforeNav) {
+                this.setState({waiting: true});
+                await beforeNav(e);
+            }
+
+            if (safeWithNav) {
+                if (!this.state.waiting) {
+                    this.setState({waiting: true});
```

Comment:
I think we only want to set this if we're doing a server-side bag, as otherwise the user won't see any impact on clicking the link.

Reply:
I think it depends.  If we're navigating from `/foo` to `/foo/bar` and the button is visible on both routes then the users will see a spinner in that situation.

Reply:
But we only want that for beforeNav, now safeWithNav. safeWithNav should only show a spinner if it's a server-side link as it doesn't block navigation when on the client.

Reply:
I thought the spinner was to indicate that stuff we were doing stuff in the background.  If we want it to indicate that we're navigating then not showing it for client-side nav with `safeWithNav` makes sense.  I'll make that change and add some comments as to why it isn't show for that case.